docs_user/modules/assembly_ospdo-scale-down-pre-database-adoption.adoc

```diff
@@ -0,0 +1,215 @@
+        enabled: true
+        dhcp: false
+      ipv6:
+        enabled: false
+    - bridge:
+        options:
+          stp:
+            enabled: false
+        port:
+        - name: enp6s0
+      description: Linux bridge with enp6s0 as a port
+      ipv4:
+        enabled: false
+      ipv6:
+        enabled: false
+      mtu: 1500
+      name: br-external
+      state: up
+      type: linux-bridge
+  nodeSelector:
+    kubernetes.io/hostname: <ostest-master-node>
+    node-role.kubernetes.io/worker: ""
+EOF
+
+oc apply -f /tmp/node3_nncp.yaml
+----
++
+. Get remaining OSPdO operands and delete everything, except DO NOT DELETE OpenStackBaremetalSets nor OpenStackProvisionServers yet
++
+----
+for i in $(oc get crd | grep osp-director | grep -v baremetalset | grep -v provisionserver | awk {'print $1'}); do echo Deleting $i...; oc delete $i -n openstack --all; done
+----
++
+. Scale down OSPdO to 0 replicas
++
+----
+ospdo_csv_ver=$(oc get csv -n openstack -l operators.coreos.com/osp-director-operator.openstack -o json | jq -r '.items[0].metadata.name')
+oc patch csv -n openstack $ospdo_csv_ver --type json   -p="[{"op": "replace", "path": "/spec/install/spec/deployments/0/spec/replicas", "value": "0"}]"
+----
++
+. Remove the webhooks from OSPdO
++
+----
+oc patch csv $ospdo_csv_ver -n openstack --type json -p="[{"op": "remove", "path": "/spec/webhookdefinitions"}]"
+----
++
+. Remove the finalizer from the OSPdO OpenStackBaremetalSet
++
+----
+oc patch openstackbaremetalsets.osp-director.openstack.org -n openstack compute --type json -p="[{"op": "remove", "path": "/metadata/finalizers"}]"
+----
++
+. Now delete the OpenStackBaremetalSet and OpenStackProvisionServer
++
+----
+oc delete openstackbaremetalsets.osp-director.openstack.org -n openstack --all
+oc delete openstackprovisionservers.osp-director.openstack.org -n openstack --all
+----
++
+Annotate all OSP compute BMHs so that Metal3 no longer handles them
++
+* Do this for each OSP compute BMH
++
+----
+compute_bmh_list=$(oc get bmh -n openshift-machine-api |grep compute|awk '{printf $1 " "}')
+
+for bmh_compute in $compute_bmh_list;do oc annotate bmh -n openshift-machine-api $bmh_compute baremetalhost.metal3.io/detached="";
+oc delete bmh -n openshift-machine-api $bmh_compute;done
+----
++
+. Delete OSPdO OLM resources (this will remove the OpenStack Director Operator)
++
+----
+oc delete subscription osp-director-operator -n openstack
+oc delete operatorgroup osp-director-operator -n openstack
+oc delete catalogsource osp-director-operator-index -n openstack
+oc delete csv $ospdo_csv_ver -n openstack
+----
++
+. Scale-up the RHOSO OpenStack Operator controller-manager to 1 replica so that the associated OpenStackControlPlane is reconciled and its OpenStackClient pod is recreated
++
+----
+oc patch csv -n openstack-operators openstack-operator.v0.0.1 --type json   -p="[{"op": "replace", "path": "/spec/install/spec/deployments/0/spec/replicas", "value": "1"}]"
+----
```

Comment:
ok , 
"reconciled", here and to OCP operators in general means: "act" according to the resources I defined,
so here, once the controller-manager operator is up and running (as we patch it to be 1 instance instead of 0), it reads its instructions in the OpenStackControlPlane resource and executes whatever is needed so that everything defined in the that OpenStackControlPlane CR is alive and in existance, specifically we pay attention here to the OpenStackClient resources as the focused part of that CR.